test/unit-test/dawn/Optimizer/Passes/TestEnvironment.h

```diff
@@ -0,0 +1,26 @@
+/*===----------------------------------------------------------------------------------*- C -*-===*\
```

Comment:
This functionality is in the wrong spot here. 

The goal of the test-directory is a mirror of the src with each file representing the file it tests. This is test-required functionality and should therefore be in src/dawn/Unittest/
I know that there are some not-so-nice ways where this has not been done in the codebase but I think we should stick to it nevertheless.

Reply:
alright, got it. 
This TestEnvironmnent is a small class that contains information specific to how we run the tests in this subproject "Passes", since they require a std::string path. 
I can move it to `src/dawn/Unittest` but it will be used specifically only by unittest in the Passes. Do you still think we should move it there? How do we handle if we need multiple of this TestEnvironmnet for different subprojects?

Reply:
ah, got it. Yeah, you're right. We can definitely keep it as is.